### PR TITLE
[MAINTENANCE] Simplify DataAssistant Operation to not Depend on Self-Initializing Expectations

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* `DataAssistant` operation (with code reduction) is simplified by making it be governed only on explicitly specified `Rule` configurations.  `Rule` `ParameterBuilder` configurations in the `ParameterBuilder` layer serve the metrics calculation purposes.  The `validation` `ParameterBuilder` objects supply `kwargs` to `ExpectationConfigurationBuilder` objects.
* Add unit tests to insure that all required metrics are emitted.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-822/GREAT-904
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/data_assistant/data_assistant.py [hunk 1]
```diff
@@ -343,69 +360,30 @@ def __init__(
 
         self._validator = validator
 
+        self._batches = self._validator.batches
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/data_assistant/data_assistant.py:363]: ❤️

File: tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py [hunk 2]
```diff
@@ -89,7 +90,38 @@ def test_onboarding_data_assistant_result_serialization(
...
+    assert num_metrics == 2
+
+    num_metrics = 0
+    for (
+        domain,
+        parameter_values_for_fully_qualified_parameter_names,
+    ) in bobby_onboarding_data_assistant_result.metrics_by_domain.items():
+        num_metrics += len(parameter_values_for_fully_qualified_parameter_names)
+
+    assert num_metrics == 178
```
Comment (NathanFarmer) [inline tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py:124]: Can we also assert that `num_metrics >= len(bobby_onboarding_data_assistant_result.expectation_configurations)`?

File: tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py [hunk 3]
```diff
@@ -89,7 +90,38 @@ def test_onboarding_data_assistant_result_serialization(
...
+            num_metrics += len(parameter_values_for_fully_qualified_parameter_names)
+
+    assert num_metrics == 2
+
+    num_metrics = 0
+    for (
+        domain,
+        parameter_values_for_fully_qualified_parameter_names,
+    ) in bobby_onboarding_data_assistant_result.metrics_by_domain.items():
+        num_metrics += len(parameter_values_for_fully_qualified_parameter_names)
```
Comment (NathanFarmer) [inline tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py:122]: Why did you need to use `domain_key` for `MetricDomainTypes.TABLE`, but not column? Do the table metrics not get double counted?

**Review by NathanFarmer — APPROVED** [review]
LGTM! Just one question about the test you added.

File: tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py:124]: @NathanFarmer number of metrics and number of expectations is generally not same.  Thanks.

File: tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py (hunk 3 above)
Comment (alexsherstinsky) [reply tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py:122]: @NathanFarmer There are many column domains; only one table domain -- so this was a "quick check".  Thanks.
